Auth48 changes (#289)
unicorn-wg/oblivious-http: draft-ietf-ohai-ohttp.md

HPKE Symmetric Algorithms Length:

: A 16-bit integer in network byte order that encodes the length, in bytes, of
  the HPKE Symmetric Algorithms field that follows.

HPKE Symmetric Algorithms:

: One or more pairs of identifiers for the different combinations of HPKE KDF
  and AEAD that the Oblivious Gateway Resource supports:
  <dl newline="true">
  <dt>HPKE KDF ID:</dt>
  <dd markdown="1">
  A 16-bit HPKE KDF identifier as defined in {{Section 7.2 of HPKE}} or [the

>   "HPKE KDF Idnetifiers" IANA

Reviewer: ```suggestion
  "HPKE KDF Identifiers" IANA
```

~~~

Clients decrypt an Encapsulated Response by reversing this process.  That is,
Clients first parse `enc_response` into `response_nonce` and `ct`. Then, they
follow the same process to derive values for `aead_key` and `aead_nonce`, using
their sending HPKE context, `sctxt`, as the HPKE context, `context`.

> The Client uses these values to decrypt `ct` using the `Open` function provided
> by the AEAD. Decrypting might produce an error, as follows:

Reviewer: ```suggestion
The Client uses these values to decrypt `ct` using the AEAD function `Open`. Decrypting might produce an error, as follows:
```

For symmetry with how we invoke the Seal function above.

information it might forward in a response MUST be encapsulated, unless it is
responding to errors that do not relate to processing the contents of the
Encapsulated Request; see {{errors}}.

An Oblivious Gateway Resource, if it receives any response from the Target
Resource, sends a single 200 response containing the Encapsulated Response.
Like the request from the Client, this response MUST only contain those fields
necessary to carry the Encapsulated Response: a 200 status code, a header field
indicating the content type, and the Encapsulated Response as the response
content.  As with requests, additional fields MAY be used to convey information
that does not reveal information about the Encapsulated Response.

An Oblivious Gateway Resource that does not receive a response can itself
generate a response with an appropriate error status code (such as 504 (Gateway
Timeout); see {{Section 15.6.5 of HTTP}}), which is then encapsulated in the
same way as a successful response.

In order to achieve the privacy and security goals of the protocol, an Oblivious
Gateway Resource also needs to observe the guidance in
{{server-responsibilities}}.

## Informational Responses

> This encapsulation does not permit the progressive processing of responses.

Reviewer: ```suggestion
This encapsulation does not permit progressive processing of responses.
```

Adding "the" here reads wrong to me.

One goal of this design is that independent Client requests are only linkable by
their content.  However, the choice of Client configuration might be used to
correlate requests.  A Client configuration includes the Oblivious Relay
Resource URI, the Oblivious Gateway key configuration, and the Oblivious Gateway
Resource URI. A configuration is active if Clients can successfully use it for
interacting with a target.

Oblivious Relay and Gateway Resources can identify when requests use the same
configuration by matching the key identifier from the key configuration or the
Oblivious Gateway Resource URI.  The Oblivious Gateway Resource might use the
source address of requests to correlate requests that use an Oblivious Relay
Resource run by the same operator.  If the Oblivious Gateway Resource is willing
to use trial decryption, requests can be further separated into smaller

> groupings based on the keys that are used.

Reviewer: ```suggestion
groupings based on active configurations that clients use.
```

Since it's not just about key usage.
